src/Symfony/Component/Messenger/Transport/RedisExt/Connection.php

```diff
@@ -157,6 +161,21 @@ public function add(string $body, array $headers)
         }
     }
 
+    public function count(): int
+    {
+        $e = null;
+        try {
+            $length = $this->connection->xlen($this->stream);
```

Comment:
xlen does actually return the length of the whole stream not messages which are in the queue. Currently I think the only way to get the length would be first do a: 

```
xinfo groups STREAMNAME
```

And then use the $this->group to get `last-delivered-id` (this could be saved into variable and set on read so we don't need todo this everytime)

and then do a:

```
xrange new2 LASTDELIVEREDID +
```

which will return an array of messages which can be counted.

Reply:
Would XPENDING be the way to move forward in getting the pending messages in a Redis stream? https://redis.io/commands/xpending/